### PR TITLE
feat: add `job_timeout_ms` to job configuration classes

### PR DESCRIPTION
**Pull request by DevStephanie**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

- correcting syntax for jobtimeoutms
- adding test to test_query, based on example comment

### CODE REVIEW

File: google/cloud/bigquery/job/base.py [hunk 1]
```diff
@@ -171,6 +171,33 @@ def __setattr__(self, name, value):
...
+        """ Optional parameter. Job timeout in milliseconds. If this time limit is exceeded, BigQuery might attempt to stop the job.
+
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/Job#JobConfiguration.FIELDS.job_timeout_ms
+
+        e.g.
+            job_config = bigquery.QueryJobConfig( jobtimeout = 5000 )
+            
+                or
+
+            job_config.jobtimeout = 5000
```
Comment (tswast) [inline google/cloud/bigquery/job/base.py:185]: Update the example's syntax to match your new property name.

File: google/cloud/bigquery/job/base.py [hunk 2]
```diff
@@ -171,6 +171,33 @@ def __setattr__(self, name, value):
...
         super(_JobConfig, self).__setattr__(name, value)
 
+    @property
+    def jobtimeout_ms(self):
+        """ Optional parameter. Job timeout in milliseconds. If this time limit is exceeded, BigQuery might attempt to stop the job.
+
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/Job#JobConfiguration.FIELDS.job_timeout_ms
+
+        e.g.
+            job_config = bigquery.QueryJobConfig( jobtimeout = 5000 )
```
Comment (tswast) [inline google/cloud/bigquery/job/base.py:181]: Update the example's syntax to match your new property name.

File: google/cloud/bigquery/job/base.py [hunk 3]
```diff
@@ -171,6 +171,33 @@ def __setattr__(self, name, value):
...
+            
+                or
+
+            job_config.jobtimeout = 5000
+        Raises:
+            ValueError: If ``value`` type is invalid.
+        """
+        
+        # None as this is an optional parameter.
```
Comment (tswast) [inline google/cloud/bigquery/job/base.py:190]: It could be `None`, but it could also be a real integer value. We should return `self._properties.get("jobTimeoutMs")` (converted to an integer if not None. I believe there are other integer properties that use a helper function for this conversion.)

File: google/cloud/bigquery/job/base.py [hunk 4]
```diff
@@ -171,6 +171,33 @@ def __setattr__(self, name, value):
...
+        return None
+
+    @jobtimeout_ms.setter
+    def jobtimeout(self, value):
+        if not isinstance(value, int):
```
Comment (tswast) [inline google/cloud/bigquery/job/base.py:195]: We should also allow `None`.

File: google/cloud/bigquery/model.py [hunk 5]
```diff
@@ -25,6 +25,7 @@
 from google.cloud.bigquery import _helpers
 from google.cloud.bigquery import standard_sql
 from google.cloud.bigquery.encryption_configuration import EncryptionConfiguration
+import google.cloud.bigquery.model
```
Comment (tswast) [inline google/cloud/bigquery/model.py:28]: We can revert the changes in this file.

File: tests/unit/job/test_query.py [hunk 6]
```diff
@@ -689,6 +690,10 @@ def test_ddl_target_table(self):
         self.assertEqual(job.ddl_target_table.dataset_id, "ddl_ds")
         self.assertEqual(job.ddl_target_table.project, self.PROJECT)
 
+    def test_jobtimeout_ms(self): 
+        self.assertIsInstance(job.timeout_ms, value)
```
Comment (tswast) [inline tests/unit/job/test_query.py:694]: Let's set the job configuration timeout ms to something before asserting the value.

File: tests/unit/job/test_query.py [hunk 7]
```diff
@@ -244,6 +244,7 @@ def test_ctor_defaults(self):
         self.assertIsNone(job.destination_encryption_configuration)
         self.assertIsNone(job.range_partitioning)
         self.assertIsNone(job.time_partitioning)
+        self.assertIsNone(job.timeout_ms) ////
```
Comment (tswast) [inline tests/unit/job/test_query.py:247]: Don't need trailing `////`

File: tests/unit/model/test_model.py [hunk 8]
```diff
@@ -349,9 +349,9 @@ def test_set_labels(object_under_test):
 
 
 def test_replace_labels(object_under_test):
-    assert object_under_test.labels == {}
-    object_under_test.labels = {"data_owner": "someteam"}
-    assert object_under_test.labels == {"data_owner": "someteam"}
+    assert object_under_test.labels() =={}
```
Comment (tswast) [inline tests/unit/model/test_model.py:352]: Revert these changes.

File: tests/unit/job/test_query.py (hunk 6 above)
Comment (tswast) [reply tests/unit/job/test_query.py:694]: Also, move these tests to where we test `QueryJobConfig`: https://github.com/googleapis/python-bigquery/blob/main/tests/unit/job/test_query_config.py#L20

**Review by tswast — CHANGES_REQUESTED** [review]

File: google/cloud/bigquery/job/base.py [hunk 9]
```diff
@@ -171,6 +171,33 @@ def __setattr__(self, name, value):
             )
         super(_JobConfig, self).__setattr__(name, value)
 
+    @property
+    def jobtimeout_ms(self):
```
[the remark this answers is not in the document]
Comment (DevStephanie) [reply google/cloud/bigquery/job/base.py:175]: Resolved on 175 and resolved on 193

File: google/cloud/bigquery/job/base.py (hunk 2 above)
Comment (DevStephanie) [reply google/cloud/bigquery/job/base.py:181]: resolved on 181, matches new property name

File: google/cloud/bigquery/job/base.py (hunk 1 above)
Comment (DevStephanie) [reply google/cloud/bigquery/job/base.py:185]: Resolved syntax on 185

File: tests/unit/job/test_query.py (hunk 7 above)
Comment (DevStephanie) [reply tests/unit/job/test_query.py:247]: Resolved, it was an accidental comment, won't happen again.

**Comment by chalmerlowe** [comment]
Thanks for providing this PR. I am grateful that you are looking to assist.

For the record: work is being conducted in the following PR to address this issue.
I submitted some ideas back to the original author of 1656 to help them advance their code and provide tests. I am gonna finalize that effort. Will close this PR.

https://github.com/googleapis/python-bigquery/pull/1656
